src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -315,6 +315,13 @@ class ClientChannel::PromiseBasedCallData : public ClientChannel::CallData {
  public:
   explicit PromiseBasedCallData(ClientChannel* chand) : chand_(chand) {}
 
+  ~PromiseBasedCallData() override {
+    if (was_queued_) {
+      MutexLock lock(&chand_->resolution_mu_);
+      chand_->resolver_queued_calls_.erase(this);
```

Comment:
I think this also needs to call `calld->RemoveCallFromResolverQueuedCallsLocked()` to clean up the pollset_set linkage.